subtest/datasets/dataset.py

```diff
@@ -68,3 +69,33 @@ def __default_train_split_local(x, y):
 
     def generate_new_model(self):
         return self.generate_new_model_for_dataset()
+
+    def shorten_dataset_proportion(self, dataset_proportion):
+        """Truncate the dataset depending on self.dataset_proportion"""
+
+        if dataset_proportion == 1:
+            raise Exception("shorten_dataset_proportion shouldn't be called on this scenario, \
+                the user targets the full dataset")
```

Comment:
Hmm, we can question that. It makes the method less resilient, as it imposes a test on the `dataset_proportion` prior to calling it. Intuitively I'd suggest handling the `== 1` case by returning the entire sets

Reply:
I approuve, I change that